src/Sylius/Component/Order/Modifier/OrderItemQuantityModifier.php

```diff
@@ -42,6 +42,8 @@ public function modify(OrderItemInterface $orderItem, $targetQuantity)
             return;
         }
 
+        $targetQuantity = min($targetQuantity, 9999);
```

Comment:
Then you can configure it by passing an argument to the constructor. 
`new LimitingOrderItemQuantityModifier($decoratedOrderItemQuantityModifier, 9999)`